api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -63,7 +63,16 @@ public Person getPerson(String id) {
 	}
 
 	@Override
-	public List<Person> searchPersonByName(String name) {
-		return null;
+	public List<Person> searchPersonsByName(String name) {
+		List<org.openmrs.Person> persons = searchPersonsByQuery(name);
+		List<Person> fhirPersonsList = new ArrayList<Person>();
+		for (org.openmrs.Person person : persons) {
+			fhirPersonsList.add(FHIRPersonUtil.generatePerson(person));
+		}
+		return fhirPersonsList;
+	}
+
+	private List<org.openmrs.Person> searchPersonsByQuery(String query) {
```

Comment:
@harsha89 I was thinking that we can use additional method as a generic method for all other methods. But you are correct, it's just an additional three line. I'll fix it.